File: EcoBookAiBackend/src/main/java/com/ecobook/dto/ReferenceDataCatalogDTO.java
package com.ecobook.dto;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class ReferenceDataCatalogDTO {
    private List<ReferenceOptionDTO> disciplinas;
    private List<ReferenceOptionDTO> niveisEnsino;
    private List<ReferenceOptionDTO> sistemasEnsino;
    private List<ReferenceOptionDTO> estadosConservacao;
    private List<ReferenceOptionDTO> necessidadesAcademicas;
}

File: EcoBookAiAndroid/app/src/main/kotlin/com/ecobook/dto/ReferenceDataCatalogDTO.kt
package com.ecobook.dto

data class ReferenceDataCatalogDTO(
    val disciplinas: List<ReferenceOptionDTO> = emptyList(),
    val niveisEnsino: List<ReferenceOptionDTO> = emptyList(),
    val sistemasEnsino: List<ReferenceOptionDTO> = emptyList(),
    val estadosConservacao: List<ReferenceOptionDTO> = emptyList(),
    val necessidadesAcademicas: List<ReferenceOptionDTO> = emptyList()
)
